### ISSUE TITLE
Feature Request Addons

### ISSUE THREAD

**Issue by Jeeppler** [terraform-aws-modules/terraform-aws-eks#19]
## I'm submitting a

* [ ] feature request

## Feature Request Addons

I would be nice to have an option which allows to install addons together with the cluster. There is another Terraform Registry module which does that: https://registry.terraform.io/modules/scholzj/kubernetes/aws/1.3.4#addons

**Comment by brandoconnor** [comment]
Right on @Jeeppler !

I've started to go down this route on a branch but was met with some difficulty controlling k8s the "terraform way" within EKS. I'll have a closer look at that module to see if there's anything I can't use from that repo. Thanks for bringing that to my attention.

From a broader standpoint, I think it might be a bit prescriptive to enable/install/embed specific pieces of software that runs on k8s through the module. I would rather make it simple for users to consume the module and add those layers on top of the outputs, and leave implementation up to downstream developers, if that makes sense. Perhaps there's a case to add more to the repo in the form of working test fixtures that show how this works (this the path I started down). I'll get back to this later in the weekend to investigate further.

Thanks again on the ideas! 💯

**Comment by brandoconnor** [comment]
Yea doing an end-to-end terraform defined k8s cluster that sets up the web UI, dashboard user, monitoring, etc. would be great. IIRC the k8s terraform provider isn't able to create a user and attach the correct policies via `rolebinding` currently so we're blocked on creating the user with an appropriate permission set. If anyone has managed to start using the EKS cluster through the terraform provider following a provision, I'm really interested to know how as the user I get seems unable to do anything. Until that's resolved, I'm somewhat stuck using `kubectl` to do anything useful to the cluster, once running.

**Comment by Jeeppler** [comment]
@max-rocket-internet thank you.

**Comment by max-rocket-internet** [comment]
Hi @Jeeppler 
For the reasons discussed in https://github.com/terraform-aws-modules/terraform-aws-eks/issues/99 I think for now we are not going to attempt to manage k8s resources from this module. This includes installing charts and add-ons. Long story short, it adds quite a lot of complexity.

As I mentioned before, helmfile is great for bulk installing charts with your custom settings. And also just helm alone will solve much of the pain for installing apps on your new cluster.

Maybe you could also experiment by using the [terraform kubernetes provider](https://www.terraform.io/docs/providers/kubernetes/index.html)

Thanks!
